src/Command/PullRequest/PullRequestMergeCommand.php

```diff
@@ -225,6 +231,44 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
     }
 
+    private function canUseApi(InputInterface $input)
+    {
+        if (true === $this->getConfig()->get('disable_merge_api', Config::CONFIG_ALL, false)) {
+            return false;
+        }
+
+        if ($input->getOption('squash') ||
```

Comment:
Any suggestions? You need to detect whether the Adapter support this operation.
But I don't like adding yet another interface like `CanSquashPullRequestOnMerge` 💀 

Adding an extra arguments on the method is no problem btw 👍